### PR TITLE
"Quote in new note" action

### PR DESCRIPTION
**Pull request by bfollington**
Fixes https://github.com/subconsciousnetwork/subconscious/issues/871

- MVP quoting feature

https://github.com/subconsciousnetwork/subconscious/assets/5009316/a0b1f522-0cf0-476c-8a3d-959e382d3fca

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Detail/DetailStackView.swift [hunk 1]
```diff
@@ -612,6 +612,14 @@ extension DetailStackAction {
...
         case let .requestAssignNoteColor(address, color):
             return .requestAssignNoteColor(address, color)
+        case let .requestQuoteInNewDetail(address):
+            return .pushDetail(
+                .editor(
+                    MemoEditorDetailDescription(
+                        fallback: "\(address.markup)\n\n"
+                    )
+                )
+            )
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Detail/DetailStackView.swift:622]: This results in an untitled editor, preloaded with the link and the cursor autofocused in a new block. 

There is one UX rough edge, if you back out of the new editor this opens, then the app will automatically save the note... which creates a "garbage" note. Perhaps we should have some clever logic that never saves a note unless it contains something other than / in addition to a single slashlink?

File: xcode/Subconscious/Shared/Components/Detail/MemoEditorDetailMetaSheetView.swift [hunk 2]
```diff
@@ -54,7 +54,26 @@ struct MemoEditorDetailMetaSheetView: View {
...
+                        
+                        if let address = store.state.address {
+                            Button(
+                                action: {
+                                    store.send(.requestQuoteInNewNote(address))
+                                },
+                                label: {
+                                    Label(
+                                        "Quote in new note",
+                                        systemImage: "quote.opening"
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Detail/MemoEditorDetailMetaSheetView.swift:68]: I welcome discussion on the best wording and icon for this action.

To me "linking" is inserting a slashlink into an existing note and "quoting" is creating a new note that references / reflects on another note (using linking).

**Comment by bmann** [comment]
Clipping might be another word. I’m clipping it into my own notebook.

(This looks good!)

**Review by gordonbrander — APPROVED** [review]
LGTM.

My one UX feedback is that I think it might be nicer to place the cursor above the link, with the link on it's own "paragraph" below? I'm imagining what this will look like with transcludes support, and by analogy to quote tweet, you probably want your thoughts above the transcluded content, not below it.

**Comment by bfollington** [comment]
> LGTM.
> 
> My one UX feedback is that I think it might be nicer to place the cursor above the link, with the link on it's own "paragraph" below? I'm imagining what this will look like with transcludes support, and by analogy to quote tweet, you probably want your thoughts above the transcluded content, not below it.

I implemented this and I think it is better, going to run with putting the cursor ahead of the quoted content.

**Pull request merged**